src/v/pandaproxy/schema_registry/json.cc

```diff
@@ -298,6 +300,454 @@ result<json::Document> parse_json(std::string_view v) {
+    // values so that "minimum" is always the inclusive limit and
+    // "exclusiveMinimum" is always the exclusive range. in this check we
+    // require for them to be the same datatype
+
+    if (!is_numeric_property_value_superset(
+          older, newer, "minimum", std::less_equal<>{})) {
+        // older["minimum"] is not superset of newer["minimum"] because newer is
+        // less strict
+        return false;
+    }
+    if (!is_numeric_property_value_superset(
+          older, newer, "maximum", std::greater_equal<>{})) {
+        // older["maximum"] is not superset of newer["maximum"] because newer
+        // is less strict
+        return false;
+    }
+
+    // TODO handle double with a big decimal lib. older["multipleOf"]: 1.1,
+    // newer["multipleOf"]: 2.2 would throw but they valid values that are
+    // compatible
+    if (
+      !is_numeric_property_value_superset(
+        older, newer, "multipleOf", [](double older, double newer) {
+            auto older_trunc = std::trunc(older);
+            auto newer_trunc = std::trunc(newer);
+
+            if (older_trunc != older || newer_trunc != newer) {
+                throw std::runtime_error{fmt::format(
+                  R"({}-multipleOf not implemented for non-integers. input: older: '{}', newer: '{}')",
+                  __FUNCTION__,
+                  older,
+                  newer)};
+            }
+            // the caller function restricts the range of values so that this
+            // cast to uint64 is safe
+            return std::gcd(uint64_t(older_trunc), uint64_t(newer_trunc))
+                   == uint64_t(older_trunc);
+        })) {
+        return false;
+    }
+
+    if (auto [maybe_gate_value, older_it, newer_it]
+        = extract_property_and_gate_check(older, newer, "exclusiveMinimum");
+        maybe_gate_value.has_value()) {
+        if (maybe_gate_value == false) {
+            return false;
+        }
+        // true: check other properties
+    } else {
+        // need to perform checks on actual values
+        if (older_it->IsBool() && newer_it->IsBool()) {
+            // both have value and can be decoded as bool
+            if (older_it->GetBool() == true && newer_it->GetBool() == false) {
+                // newer has a less strict range then older
+                return false;
+            }
+            // keep checking
+        } else if (older_it->IsDouble() && newer_it->IsDouble()) {
+            // TODO extend this for double
+            throw std::runtime_error{fmt::format(
+              R"({}-exclusiveMinimum not implemented for types other than "boolean". input: older: '{}', newer: '{}')",
+              __FUNCTION__,
+              pj{older},
+              pj{newer})};
+        } else {
+            // types changes are always not compatible (one is boolean and the
+            // other is double)
+            return false;
+        }
+    }
+
+    if (auto [maybe_gate_value, older_it, newer_it]
```

Comment:
nit: there's some code duplication here across `exclusiveMinimum` and `exclusiveMaximum`. I think the only difference is the check on the boolean values and the error message. Reusing the rest of the code would make it clearer that that's the only difference.